src/bar-chart.js

```diff
@@ -147,8 +160,17 @@ dc.barChart = function (parent, chartGroup) {
         dc.transition(bars, _chart.transitionDuration())
             .attr('x', function (d) {
                 var x = _chart.x()(d.x);
-                if (_centerBar) {
-                    x -= _barWidth / 2;
+                if (_groupBars) {
+                    x += _chart.groupGap() / 2;
+                    x += layerIndex * (_barWidth + _gap);
+                    x += _gap / 2;
+                }
+                if (_centerBar && !_chart.isOrdinal()) {
```

Comment:
this check; _chart.isOrdinal() affects bar charts in general.

I saw that bar charts with ordinal x axis are already centered around axis points. So setting centerBar(true) only seems to offset it incorrectly.

Please see if this is fine